tools/apigee-envoy-quickstart/extension/README.md

```diff
@@ -0,0 +1,156 @@
+    ```bash
+    export $ISTIO_GATEWAY_NS=<namespace-hosting-ingressgateway>
+    ```  
+
+1. ### Execute the below if istio-ingressgateway is missing in the istio enabled cluster
+    ```bash
+    git clone https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages.git \
+    /tmp/asm
+
+    kubectl create namespace $ISTIO_GATEWAY_NS
+
+    kubectl label namespace $ISTIO_GATEWAY_NS istio-injection=enabled istio.io/rev-
+
+    kubectl apply -n $ISTIO_GATEWAY_NS \
+        -f /tmp/asm/samples/gateways/istio-ingressgateway
+
+    kubectl rollout restart deployment -n $ISTIO_GATEWAY_NS
+
+    kubectl get svc istio-ingressgateway -n $ISTIO_GATEWAY_NS
+    ```
+
+1. **Overwrite the service details that has to be exposed outside the mesh, if different from the service (httpbin application) deployed in the [starter kit](https://github.com/apigee/devrel/tree/main/tools/apigee-envoy-quickstart#envoy-with-apigee-adapter-as-containers-within-kubernetes-platform)**
+    ```bash
+    export TARGET_HOST="httpbin.org"
+    export TARGET_SERVICE_NAME="httpbin"
+    export TARGET_SERVICE_NAMESPACE="apigee"
+    ```
+
+1. **Fetch the Loadbalancer details exposing the ingress gatway**
+    ```bash
+    export INGRESS_HOST=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    export INGRESS_PORT=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].port}')
+    export SECURE_INGRESS_PORT=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
+    export TCP_INGRESS_PORT=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="tcp")].port}')
+    ```
+
+1. **Setup Gateway component**
+    ```bash
+   cat <<EOF > /tmp/httpbin-gateway.tmpl              
+   apiVersion: networking.istio.io/v1alpha3
+   kind: Gateway
+   metadata:
+     name: httpbin-gateway
+   spec:
+     selector:
+       istio: ingressgateway
+     servers:
+       - port:
+           number: 80
+           name: http
+           protocol: HTTP
+         hosts:
+           - "$TARGET_HOST"
+   EOF
+
+   envsubst < /tmp/httpbin-gateway.tmpl > /tmp/httpbin-gateway.yaml
+
+   kubectl apply -n $ISTIO_GATEWAY_NS -f /tmp/httpbin-gateway.yaml
+    ```
+
+1. **Configure routes for traffic entering via the Gateway:**
+    ```bash
+   cat <<EOF > /tmp/httpbin-virtual-service.tmpl
+   apiVersion: networking.istio.io/v1alpha3
+   kind: VirtualService
+   metadata:
+     name: httpbin-ingress
+   spec:
+     hosts:
+     - "$TARGET_HOST"
+     gateways:
+     - httpbin-gateway
+     http:
+     - route:
+       - destination:
+           host: $TARGET_SERVICE_NAME.$TARGET_SERVICE_NAMESPACE.svc.cluster.local
+           port:
+             number: 80
+   EOF
+
+   envsubst < /tmp/httpbin-virtual-service.tmpl > /tmp/httpbin-virtual-service.yaml
```

Comment:
you could get the variables directly in the cat. no need for cat + envsubst

Reply:
duh....